src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -158,7 +158,7 @@ def generate_samples(self):
         Returns:
              array[complex]: array of samples in the shape ``(dev.shots, dev.num_wires)``
         """
-        return [self._result_to_sample_output(res) for res in self.samples.measurements]
+        return np.array([self._result_to_sample_output(res) for res in self.samples.measurements])
```

Comment:
What is the logic here, what was the problem without this line? We usually try to avoid explicitly wrapping data in numpy arrays, so if we can get to the bottom of the problem we might be able to fix it more constructively?

Reply:
It's about what data structure `QubitDevice` expects and how indices of data are retrieved in those functions. We can fix `QubitDevice`, or we can give it the format it wants 😅

Reply:
oh damn, dont want to mess with `QubitDevice` 😬